extensions/platform-http/runtime/src/main/java/org/apache/camel/quarkus/component/platform/http/runtime/QuarkusPlatformHttpConsumer.java

```diff
@@ -0,0 +1,376 @@
+            final StringWriter sw = new StringWriter();
+            final PrintWriter pw = new PrintWriter(sw);
+            exception.printStackTrace(pw);
+
+            // the body should then be the stacktrace
+            body = ByteBuffer.wrap(sw.toString().getBytes(StandardCharsets.UTF_8));
+            // force content type to be text/plain as that is what the stacktrace is
+            message.setHeader(Exchange.CONTENT_TYPE, "text/plain; charset=utf-8");
+
+            // and mark the exception as failure handled, as we handled it by returning it as the response
+            ExchangeHelper.setFailureHandled(exchange);
+        }
+
+        // set the content type in the response.
+        final String contentType = MessageHelper.getContentType(message);
+        if (contentType != null) {
+            // set content-type
+            response.putHeader("Content-Type", contentType);
+        }
+        return body;
+    }
+
+    void writeResponse(RoutingContext ctx, Exchange camelExchange, HeaderFilterStrategy headerFilterStrategy) {
+        final Object body = toHttpResponse(ctx.response(), camelExchange.getMessage(), headerFilterStrategy);
+
+        final HttpServerResponse response = ctx.response();
+        if (body == null) {
+            QuarkusPlatformHttpEngine.LOG.tracef("No payload to send as reply for exchange: %s", camelExchange);
+            response.putHeader("Content-Type", "text/plain; charset=utf-8");
+            response.end("No response available");
+            return;
+        }
+
+        if (body instanceof String) {
+            response.end((String) body);
+        } else if (body instanceof InputStream) {
+            final byte[] bytes = new byte[4096];
+            try (InputStream in = (InputStream) body) {
+                int len;
+                while ((len = in.read(bytes))  >= 0) {
+                    final Buffer b = Buffer.buffer(len);
+                    b.appendBytes(bytes, 0, len);
+                    response.write(b);
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            response.end();
+        } else {
+            final TypeConverter tc = camelExchange.getContext().getTypeConverter();
+            try {
+                final ByteBuffer bb = tc.mandatoryConvertTo(ByteBuffer.class, body);
+                final Buffer b = Buffer.buffer(bb.capacity());
+                b.setBytes(0, bb);
+                response.end(b);
+            } catch (TypeConversionException | NoTypeConversionAvailableException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+    }
+
+    Exchange toExchange(RoutingContext ctx, Exchange exchange, HeaderFilterStrategy headerFilterStrategy) {
+        Message in = toCamelMessage(ctx, exchange, headerFilterStrategy);
+        final String charset = ctx.parsedHeaders().contentType().parameter("charset");
```

Comment:
What if this is null, then its likely better to not set these headers

Reply:
Solved in 7c292ebc4ee38f0bda8ed182c29d3cdc3957a0dd